python/pyspark/ml/evaluation.py

```diff
@@ -365,18 +375,25 @@ class MulticlassClassificationEvaluator(JavaEvaluator, HasLabelCol, HasPredictio
                  "The beta value used in weightedFMeasure|fMeasureByLabel."
                  " Must be > 0. The default value is 1.",
                  typeConverter=TypeConverters.toFloat)
+    eps = Param(Params._dummy(), "eps",
```

Comment:
Do we really need this? it's kind of arbitrary. Do other libs have an epsilon for computing log loss? if a classifier is returning 1 or 0 the log loss really is undefined.

Reply:
SKLearn has this param (default value is 1e-15), I follow sklearn's design.
Other lib like xgboost also use an epsilon to compute [log-loss](https://github.com/dmlc/xgboost/blob/master/src/objective/regression_loss.h#L72), but it's not exposed to end user and its default value is 1e-16.

Reply:
OK. I guess that's good enough reason to expose it. It's so niche that I almost think it's not worth it, but ...